flocker/node/_docker.py

```diff
@@ -527,6 +568,34 @@ def _address_in_use(self, apierror):
             return None
         return AddressInUse(address=(ip, int(port)), apierror=apierror)
 
+    def _image_data(self, image, container, state):
+        cached_data = self._image_cached_data(image)
+        if cached_data is not None:
+            return cached_data
+        try:
+            image_data = self._client.inspect_image(image)
+            self._image_cache[image] = image_data
```

Comment:
The data will end up using up a bunch of memory if there's lots of images used. I suggest only caching the information we actually care about.